### DISCUSSION TITLE
WinUI 3 Support And Improvement.

### DISCUSSION THREAD

**Discussion by sakib1361** [SuRGeoNix/Flyleaf#312] [Show and tell]
Hello, thank you for a very nice media player. I have partially modified the WINUI 3 project and integrated with my personal use media player which can play Local Library (Like Plex), Torrent and YouTube. 
In my journey, I have found that the Fullscreen container created more problem than solved. So, I kept the flyleaf host and modifed container in a way so that on Fullscreen it replaces window content with flyleaf host and keep maincontent on a temp object in memory. On exit fullscreen, detach flyleafhost and put it back at container. Now put the temp object as maincontent on window. 
I have also found that dpi scaling was no necessary and arbitrarily made the video zoomed. 
Transparent panels worked too. (Yes the image is from WinUI)
<img width="960" alt="image" src="https://github.com/SuRGeoNix/Flyleaf/assets/20419201/567dcc73-a292-47c3-baef-30e9b171e877">
I was planning on making it crossplatform. But it became way more difficult with MAUI. Microsoft is taking good tech and making it waste. So winui 3 is for now. 
I'll try to fork this project and try to implement my changes there. (Only the winui).

**Comment by SuRGeoNix** [comment]
HI @sakib1361,
Tbh, didn't test much for WinUI 3 as I was really disappointed from Microsoft. You actually need to write more Win32 API code rather than C#. It still has a lot of issues (including the swapchainpanel). Feel free to create a PR!
For the cross-platform I will need to first to update the FlyleafLib to support it (planning to do this in v4 which I will re-code from scratch).
